apps/ejabberd/src/ejabberd_c2s.erl

```diff
@@ -875,6 +875,9 @@ session_established({xmlstreamelement,
 session_established({xmlstreamelement, El}, StateData) ->
     FromJID = StateData#state.jid,
     % Check 'from' attribute in stanza RFC 3920 Section 9.1.2
+    % this is where we probably should initialise accumulator, or even earlier
+    % but it would require hacking mod_amp at the very beginning
+    % so to make it simpler we do it a bit later
```

Comment:
Yes :) Here, "later" means in the next function which is `process_outgoing_stanza`